File: api-stockhub/stockhub/src/main/java/com/api/stockhub/dtos/ColaboradorRecordDto.java
package com.api.stockhub.dtos;

import jakarta.validation.constraints.NotBlank;
import jakarta.validation.constraints.NotNull;

public record ColaboradorRecordDto( Long matricula, @NotBlank String nome, @NotBlank String empresa,
		@NotBlank String licenca, @NotBlank String funcao, @NotBlank String setor) {

}

File: StockHub/StockHub/Domain/DTOs/ColaboradorRecordDto.cs
using System.ComponentModel.DataAnnotations;
namespace StockHub.Domain.DTOs
{
   public record ColaboradorRecordDto(
        long Matricula,
        [Required] string Nome,
        [Required] string Empresa,
        [Required] string Licenca,
        [Required] string Funcao,
        [Required] string Setor
    );
}
